File: lemur-app/src/main/java/io/github/malczuuu/lemur/app/domain/player/PlayerEventGateway.java
package io.github.malczuuu.lemur.app.domain.player;

public interface PlayerEventGateway {

  void publish(PlayerEvent.PlayerCreated event);

  void publish(PlayerEvent.PlayerUpdated event);

  void publish(PlayerEvent.PlayerBanned event);

  void publish(PlayerEvent.PlayerUnbanned event);

  void publish(PlayerEvent.PlayerRatingChanged event);
}

File: lemur-app/src/main/kotlin/io/github/malczuuu/lemur/app/domain/player/PlayerEventGateway.kt
package io.github.malczuuu.lemur.app.domain.player

interface PlayerEventGateway {

    fun publish(event: PlayerEvent.PlayerCreated)

    fun publish(event: PlayerEvent.PlayerUpdated)

    fun publish(event: PlayerEvent.PlayerBanned)

    fun publish(event: PlayerEvent.PlayerUnbanned)

    fun publish(event: PlayerEvent.PlayerRatingChanged)
}
